magicctapipe/utils/functions.py

```diff
@@ -1,294 +1,317 @@
+        z_coord_mean = group_sum["weighted_z_coord"] / group_sum["weight"]
 
     coord_mean = SkyCoord(
-        x=x_coord_mean.values,
-        y=y_coord_mean.values,
-        z=z_coord_mean.values,
+        x=x_coord_mean.to_numpy(),
+        y=y_coord_mean.to_numpy(),
+        z=z_coord_mean.to_numpy(),
         representation_type="cartesian",
     )
 
-    lon_mean = coord_mean.spherical.lon
-    lat_mean = coord_mean.spherical.lat
+    # Transform the cartesian to the spherical coordinate
+    coord_mean = coord_mean.spherical
+
+    lon_mean = pd.Series(data=coord_mean.lon.to_value(unit), index=x_coord_mean.index)
+    lat_mean = pd.Series(data=coord_mean.lat.to_value(unit), index=x_coord_mean.index)
 
     return lon_mean, lat_mean
 
 
-def calculate_angular_distance(on_coord, event_coord, tel_coord, n_off_regions):
+@u.quantity_input
+def calculate_off_coordinates(
+    pointing_ra: u.deg,
+    pointing_dec: u.deg,
+    on_coord_ra: u.deg,
+    on_coord_dec: u.deg,
+    n_off_regions,
+):
     """
-    Calculates the angular distance between
-    the shower arrival direction and ON/OFF regions.
+    Calculates the coordinates of the centers of OFF regions to estimate
+    the backgrounds for wobble observation data.
+
+    It calculates the wobble offset and rotation angle with equations
+    derived from a hand calculation.
 
     Parameters
     ----------
-    on_coord: astropy.coordinates.sky_coordinate.SkyCoord
-        Coordinate of the ON region
-    event_coord: astropy.coordinates.sky_coordinate.SkyCoord
-        Coordinate of the shower arrival direction
-    tel_coord: astropy.coordinates.sky_coordinate.SkyCoord
-        Coordinate of the telescope pointing direction
+    pointing_ra: astropy.units.quantity.Quantity
+        Right ascension of the telescope pointing direction
+    pointing_dec: astropy.units.quantity.Quantity
+        Declination of the telescope pointing direction
+    on_coord_ra: astropy.units.quantity.Quantity
+        Right ascension of the center of the ON region
+    on_coord_dec: astropy.units.quantity.Quantity
+        Declination of the center of the ON region
     n_off_regions: int
         Number of OFF regions to be extracted
 
     Returns
     -------
-    theta_on: astropy.units.quantity.Quantity
-        Angular distance from the ON region
-    theta_off: dict
-        Angular distances from the OFF regions
     off_coords: dict
-        Coordinates of the OFF regions
+        Coordinates of the centers of the OFF regions
     """
 
-    # Compute the distance from the ON region:
-    theta_on = on_coord.separation(event_coord)
+    ra_diff = pointing_ra - on_coord_ra
 
-    # Compute the wobble offset and rotation:
-    offsets = np.arccos(
-        np.cos(on_coord.dec)
-        * np.cos(tel_coord.dec)
-        * np.cos(tel_coord.ra - on_coord.ra)
-        + np.sin(on_coord.dec) * np.sin(tel_coord.dec)
+    # Calculate the wobble offset
+    wobble_offset = np.arccos(
```

Comment:
why not simply use the angular distance function?

Reply:
I guess you mean the `astropy.coordinates.angular_separation` function, so I changed the script to use it. I confirmed that the result is consistent with the one obtained by the equation.